integration/upload_test.go

```diff
@@ -0,0 +1,257 @@
+}
+
+func (suite *fileUploadSuite) SetupSuite() {
+	suite.Setup(suite.T())
+
+	opts := env.Options{RequiredIfNoDef: false}
+	suite.uploadCfg = uploadTestConfig{}
+	require.NoError(suite.T(), env.Parse(&suite.uploadCfg, opts), "failed to process upload environment variables")
+	suite.T().Logf("upload test configuration - %v", suite.uploadCfg)
+	suite.checkUploadDir()
+
+	fmt.Println(suite.ThingCfg)
+	suite.thingURL = util.GetThingURL(suite.Cfg.DigitalTwinAPIAddress, suite.ThingCfg.DeviceID)
+	suite.featureURL = util.GetFeatureURL(suite.thingURL, featureID)
+}
+
+func (suite *fileUploadSuite) TearDownSuite() {
+	suite.TearDown()
+}
+
+func TestFileUploadHTTP(t *testing.T) {
+	suite.Run(t, new(httpFileUploadSuite))
+}
+
+func TestFileUploadAzure(t *testing.T) {
+	suite.Run(t, new(azureFileUploadSuite))
+}
+
+func TestFileUploadAWS(t *testing.T) {
+	suite.Run(t, new(awsFileUploadSuite))
+}
+
+func (suite *httpFileUploadSuite) TestFileUpload() {
+	if len(suite.uploadCfg.HTTPServer) == 0 {
+		suite.T().Fatal("http server must be set")
+	}
+	upload := newHTTPUpload(suite.T(), suite.uploadCfg.HTTPServer)
+	suite.testUpload(upload)
+}
+
+func (suite *azureFileUploadSuite) TestFileUpload() {
+	creds, err := uploaders.GetAzureTestCredentials()
+	require.NoError(suite.T(), err, "please set azure environment variables.")
+	options, err := uploaders.GetAzureTestOptions(creds)
+	require.NoError(suite.T(), err, "error getting azure test options")
+	upload := newAzureUpload(suite.T(), options)
+	suite.testUpload(upload)
+}
+
+func (suite *awsFileUploadSuite) TestFileUpload() {
+	creds, err := uploaders.GetAWSTestCredentials()
+	require.NoError(suite.T(), err, "please set aws environment variables.")
+	options := uploaders.GetAWSTestOptions(creds)
+	upload, err := newAWSUpload(suite.T(), options)
+	require.NoError(suite.T(), err, "error creating AWS client")
+	suite.testUpload(upload)
+}
+
+func (suite *fileUploadSuite) triggerUploads() map[string]string {
+	topicRequest := util.GetLiveMessageTopic(suite.ThingCfg.DeviceID, protocol.ActionRequest)
+	pathRequest := util.GetFeatureOutboxMessagePath(featureID, string(protocol.ActionRequest))
+
+	connMessages, err := util.NewDigitalTwinWSConnection(suite.Cfg)
+	require.NoError(suite.T(), err, msgFailedCreateWebsocketConnection)
+	defer connMessages.Close()
+
+	util.SubscribeForWSMessages(suite.Cfg, connMessages, typeMessages, fmt.Sprintf(eventFilterTemplate, pathRequest))
+	_, err = util.ExecuteOperation(suite.Cfg, suite.featureURL, operationTrigger, nil)
+	require.NoErrorf(suite.T(), err, msgErrorExecutingOperation, operationTrigger)
+	requests := []interface{}{}
+	err = util.ProcessWSMessages(suite.Cfg, connMessages,
+		func(msg *protocol.Envelope) (bool, error) {
+			if msg.Topic.String() == topicRequest && msg.Path == pathRequest {
+				requests = append(requests, msg.Value)
+				return len(requests) == uploadFilesCount, nil
+			}
+			return false, fmt.Errorf(msgUnexpectedValue, msg.Value)
```

Comment:
Do we expect such unknown value? If not, do we need to stop the event processing?